homework/index.js

```diff
@@ -30,17 +15,123 @@
+    const parent = createAndAppend('tr', tbody);
+    createAndAppend('td', parent, {
+      text: `${textKey}: `,
+      class: 'label',
+    });
+    createAndAppend('td', parent, {
+      text: textContent,
+    });
+  }
+  // table of repository's basic information
+  function createDetailsTable(repository, container) {
+    const repositoryData = createAndAppend('div', container, { class: 'left-div white-frame' });
+    const table = createAndAppend('table', repositoryData);
+    const tbody = createAndAppend('tbody', table);
+    const tr = createAndAppend('tr', tbody);
+    createAndAppend('td', tr, {
+      text: 'Repository :',
+      class: 'label',
+    });
+    // manually create td with a link
+    const td = createAndAppend('td', tr);
+    createAndAppend('a', td, {
+      href: repository.html_url,
+      text: repository.name,
+      target: '_blank',
+    });
+    createTd('Description', repository.description, tbody);
+    createTd('Forks', repository.forks, tbody);
+    createTd('Updated', repository.updated_at, tbody);
+  }
+
+  // contributors list
+  async function listContributors(repository, container) {
+    const contributorUrl = repository.contributors_url;
+    const response = await fetch(contributorUrl);
+    const contributors = await response.json();
+    try {
+      const contributorsData = createAndAppend('div', container, {
+        class: 'right-div white-frame',
+      });
+      createAndAppend('p', contributorsData, {
+        text: 'contributions',
+        class: 'contributor-header',
+      });
+      const ul = createAndAppend('ul', contributorsData, {
+        class: 'contributor-list',
+      });
+      contributors.forEach(contributor => {
+        const li = createAndAppend('li', ul);
+        const anchor = createAndAppend('a', li, {
+          href: contributor.html_url,
+          target: '_blank',
+          class: 'contributor-item',
+        });
+        createAndAppend('img', anchor, {
+          src: contributor.avatar_url,
+          alt: "contributor's profile",
+          class: 'contributor-avatar',
+        });
+        const liDiv = createAndAppend('div', anchor, { class: 'contributor-data' });
+        createAndAppend('div', liDiv, { text: contributor.login, class: 'contributor-name' });
+        createAndAppend('div', liDiv, {
+          text: contributor.contributions,
+          class: 'contributor-badge',
+        });
+      });
+    } catch (err) {
+      createAndAppend('div', document.getElementById('root'), {
+        text: err.message,
+        class: 'alert-error',
+      });
+    }
+  }
+
+  // create the select options and the event listener for changing repositories
+  function selectAndChange(repositories, container, header) {
+    // localeCompare enables a case-insensitive sort of an array.
+    repositories.sort((a, b) => a.name.localeCompare(b.name));
+
+    const select = createAndAppend('select', header, { class: 'repository-selector' });
```

Comment:
You can add the `autofocus` attribute so that the `<select>` element gets the keyboard focus when the page is loaded:

```js
const select = createAndAppend('select', header, {
  class: 'repository-selector',
  autofocus: true,
});
```

Reply:
To move between the select choices you mean!
When I press `tab` it starts from the `select` already.

Reply:
With `autofocus` you don't need to press `tab` at all. You can straight away use the up and down keys to select a repo.

Reply:
Good to know, 
Thanks!